### PR TITLE
PLTFRM-1078 Add SDS filter for users without MFA count

### PR DESCRIPTION
**Pull request by brunobasto**
## Description

This PR implements SDS (Site Details Service) integration to track the count of users without Multi-Factor Authentication (MFA) enabled. 

### Added
- Added SDS filter for users without MFA count to enable monitoring of 2FA adoption across VIP environments
- Added comprehensive test coverage for SDS payload integration including edge cases for skipped users and bot exclusions

## Pre-review checklist

- [x] This change works and has been tested locally or in Codespaces (or has an appropriate fallback).
- [x] This change has relevant unit tests (if applicable).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [x] I've created a changelog description that aligns with the provided examples.

## Pre-deploy checklist

- [ ] VIP staff: Ensure any alerts added/updated conform to internal standards (see internal documentation). 

## Steps to Test

1. Check out the PR branch
2. Ensure the WordPress environment has the Two-Factor plugin active
3. Create test users with different roles (admin, editor) and MFA configurations
4. Apply the `vip_site_details_index_security_boost_data` filter:
   ```php
   $test_data = apply_filters('vip_site_details_index_security_boost_data', []);
   ```
5. Verify that `$test_data['users_without_2fa_count']` contains the correct count of users without MFA
6. Run the unit tests to verify all edge cases work correctly:
   ```bash
   composer test -- --filter=test_vip_site_details_index_security_boost_data
   ```
7. Verify that skipped users and bot users are properly excluded from the count

### CODE REVIEW

**Comment by brunobasto** [comment]
@andrea-sdl I found that user queries with `blog_id=0` and `role__in` or `capability__in` don't work as expected. Seems like it doesn't respect the capabilities or roles filters when blog_id is zero. The same problem might be happening to the inactive users counting as well. Ideas?

I'll do more tests tomorrow, maybe I'm doing something wrong, but I noticed subscribers counted in my unit tests when getting the count for blog_id=0, so I need to investigate more.

File: modules/highlight-mfa-users/class-highlight-mfa-users.php [hunk 1]
```diff
@@ -89,6 +89,29 @@ public static function register_hooks() {
...
+	public static function add_users_without_2fa_count_to_sds_payload( $data ) {
+		// Add fix for unreliable FOUND_ROWS() query
+		add_filter( 'found_users_query', [ Users_Query_Utils::class, 'fix_found_users_query' ], 10, 2 );
+
+		$users_without_2fa_count = self::get_mfa_disabled_count();
+
+		$data['users_without_2fa_count'] = $users_without_2fa_count;
+
+		// Remove fix for unreliable FOUND_ROWS() query
+		remove_filter( 'found_users_query', [ Users_Query_Utils::class, 'fix_found_users_query' ], 10 );
```
[the remark this answers is not in the document]
Comment (brunobasto) [reply modules/highlight-mfa-users/class-highlight-mfa-users.php:120]: it uses the same priority

File: modules/inactive-users/class-inactive-users.php [hunk 2]
```diff
@@ -131,6 +131,9 @@ public static function add_inactive_users_count_to_sds_payload( $data ) {
 		// Register query time metric
 		do_action( 'vip_security_inactive_users_query_time', $timer );
 
+		// Remove fix for unreliable FOUND_ROWS() query
+		remove_filter( 'found_users_query', [ Users_Query_Utils::class, 'fix_found_users_query' ], 10 );
```
[the remark this answers is not in the document]
Comment (brunobasto) [reply modules/inactive-users/class-inactive-users.php:135]: we do add it first on this same method

**Comment by brunobasto** [comment]
~I left the network-wide count (`users_without_2fa_count_all_blogs`) out on purpose, as per my tests it's not simply a matter of querying with `blog_id=0`. See [source code](https://github.com/WordPress/WordPress/blob/52a06d5ed4dfd0a2ff94231f6eb9afe57b417f1f/wp-includes/class-wp-user-query.php#L557) where we skip roles and capability filtering when `$blog_id` is a falsy value.~

~I'll work on getting the network-wide counts working on a follow-up PR~

Done. Added in https://github.com/Automattic/vip-security-boost/pull/72/commits/012b3ef42953e1acf4298500c9148c1c5f39fb7d

File: modules/highlight-mfa-users/class-highlight-mfa-users.php [hunk 3]
```diff
@@ -89,6 +89,37 @@ public static function register_hooks() {
...
+		if ( is_multisite() ) {
+			// Get number of users without 2FA for all blogs (network-wide with blog_id = 0)
+			$users_without_2fa_count_all_blogs = self::get_mfa_disabled_count( 0 );
+
+			// Add network-wide users without 2FA count to the SDS payload
+			$data['users_without_2fa_count_all_blogs'] = $users_without_2fa_count_all_blogs;
+		}
+
+		// Remove fix for unreliable FOUND_ROWS() query
+		remove_filter( 'found_users_query', [ Users_Query_Utils::class, 'fix_found_users_query' ], 10 );
```
Comment (kat3samsin) [inline modules/highlight-mfa-users/class-highlight-mfa-users.php:120]: I think we should pass the priority here to remove the filter completely.

```suggestion
		remove_filter( 'found_users_query', [ Users_Query_Utils::class, 'fix_found_users_query' ], 10, 2 );
```

File: modules/inactive-users/class-inactive-users.php (hunk 2 above)
Comment (kat3samsin) [inline modules/inactive-users/class-inactive-users.php:135]: Also here..

**Review by kat3samsin — COMMENTED** [review]
Looks good! I added a couple of comments inline.

File: modules/highlight-mfa-users/class-highlight-mfa-users.php (hunk 3 above)
Comment (brunobasto) [reply modules/highlight-mfa-users/class-highlight-mfa-users.php:120]: I'm already passing the priority. What I'm not passing is the argument count, which [remove_filter](https://developer.wordpress.org/reference/functions/remove_filter/) doesn't accept as an argument.

Comment (kat3samsin) [reply modules/highlight-mfa-users/class-highlight-mfa-users.php:120]: You're absolutely right, I miscounted. 😬

**Review by kat3samsin — APPROVED** [review]
🥇
